subsys/bluetooth/host/mesh/access.c

```diff
@@ -97,6 +97,62 @@ s32_t bt_mesh_model_pub_period_get(struct bt_mesh_model *mod)
 	return period >> mod->pub->period_div;
 }
 
+static void publish_sent(int err, void *user_data)
+{
+	struct bt_mesh_model *mod = user_data;
+	s32_t delay;
+
+	BT_DBG("err %d", err);
+
+	if (mod->pub->count) {
+		delay = BT_MESH_PUB_TRANSMIT_INT(mod->pub->retransmit);
+	} else {
+		delay = bt_mesh_model_pub_period_get(mod);
+	}
+
+	BT_DBG("Publishing next time in %dms", delay);
+
+	k_delayed_work_submit(&mod->pub->timer, delay);
+}
+
+static const struct bt_mesh_send_cb pub_sent_cb = {
+	.end = publish_sent,
+};
+
+static int publish_retransmit(struct bt_mesh_model *mod)
+{
+	struct bt_mesh_app_key *key;
+	struct bt_mesh_msg_ctx ctx = {
+		.addr = mod->pub->addr,
+		.send_ttl = mod->pub->ttl,
+	};
+	struct bt_mesh_net_tx tx = {
+		.ctx = &ctx,
+		.src = mod->elem->addr,
+		.xmit = bt_mesh_net_transmit_get(),
+		.friend_cred = mod->pub->cred,
+	};
+
+	key = bt_mesh_app_key_find(mod->pub->key);
+	if (!key) {
+		return -EADDRNOTAVAIL;
+	}
+
+	tx.sub = bt_mesh_subnet_get(key->net_idx);
+	if (tx.sub->kr_phase == BT_MESH_KR_PHASE_2 && key->updated) {
+		tx.aid = key->keys[1].id;
+	} else {
+		tx.aid = key->keys[0].id;
+	}
+
+	mod->pub->count--;
+
+	ctx.net_idx = key->net_idx;
+	ctx.app_idx = key->app_idx;
+
+	return bt_mesh_trans_resend(&tx, mod->pub->msg, &pub_sent_cb, mod);
```

Comment:
Just realize this is not going to work since the sequence number is used for constructing the AppNonce. I.e. we'll have to do the full app + net encryption for each retransmission,